hermes/views.py

```diff
@@ -1,3 +1,214 @@
+    # the environment as k8s secrets
+    username = os.getenv('HOP_USERNAME', None)
+    password = os.getenv('HOP_PASSWORD', None)
+    if username is None or password is None:
+        error_message = 'Supply Hop credentials: set HOP_USERNAME and HOP_PASSWORD environment variables.'
+        logger.error(error_message)
+        return Response({'message': 'Hop credentials are not set correctly on the server'},
+                        status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+    hop_auth = Auth(username, password)
+
+    try:
+        topic = 'hermes.test'
+        stream = Stream(auth=hop_auth)
+        # open for write ('w')
+        with stream.open(f'kafka://kafka.scimma.org/{topic}', 'w') as s:
+            metadata = {'topic': topic}
+            s.write(message, metadata)
+    except Exception as e:
+        return Response({'message': f'Error posting message to kafka: {e}'},
+                        status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+    return Response({"message": {"Message was submitted successfully"}}, status=status.HTTP_200_OK)
+
+
+class HopSubmitView(APIView):
+    """
+    Submit a message to the hop client
+    """
+    @csrf_exempt
+    def post(self, request, *args, **kwargs):
+        # what's going on here?
+        logger.info(f'args: {args}')
+        logger.info(f'kwargs: {kwargs}')
+        logger.info(f'dir(request): {dir(request)}')
+        logger.info(f'request: {request}')
+        logger.info(f'request.POST: {dir(request.POST)}')
+        logger.info(f'type(request.body): {type(request.body)}')
+        logger.info(f'request.body: {request.body}')
+
+        # extract the message JSON from the HTTPRequest
+        try:
+            message = json.loads(request.body.decode("utf-8"))
+            logger.info(f'message: {message}')
+        except json.JSONDecodeError as err:
+            error_message = f'JSONDecodeError: {err} for request body: {request.body}'
+            logger.error(error_message)
+            return Response({'message': error_message}, status=status.HTTP_400_BAD_REQUEST)
+
+        return submit_to_hop(message)
+
+    def get(self, request, *args, **kwargs):
+        return Response({"message": "Supply any valid json to send a message to kafka."}, status=status.HTTP_200_OK)
+
+
+class HopSubmitCandidatesView(APIView):
+    def get(self, request, *args, **kwargs):
+        message = """This endpoint is used to send a message with a list of potential candidates corresponding to a 
+        non-localized event.
+        
+        Requests should be structured as below:
+        
+        {title: <Title of the message>,
+         topic: <kafka topic to post message to>, 
+         author: <submitter of the message>,
+         message_text: <Text of the message to send>,
+         event_id: <ID of the non-localized event for these candidates>,
+         data: {authors: <Text full list of authors on a message>,
+                main_data: {[{candidate_id: <ID of the candidate>,
```

Comment:
The name of the main data table should be specific to the table made from the frontend, and can generally be anything you want and will be displayed as such in the message details.

Reply:
this should probably be `candidate_data` rather than `main_data` now.